firmware/port.c

```diff
@@ -707,7 +707,16 @@ void port_handle_extint(PortData *p, u32 flags) {
             if (port_pin_supports_interrupt(p, pin)) {
                 Pin sys_pin = p->port->gpio[pin];
                 if (flags & (1 << pin_extint(sys_pin))) {
-                    port_send_status(p, REPLY_ASYNC_PIN_CHANGE_N + pin);
+                    u8 response = REPLY_ASYNC_PIN_CHANGE_N + pin;
+
+                    // If the pin's state is high, set bit 3
+                    // of the response byte. This will be used to
+                    // inform "change" listeners of the present state
+                    // at the moment of the interrupt.
+                    if (pin_read(port_selected_pin(p))) {
```

Comment:
Currently, when you do a `.read()` inside of the callback for an interrupt event such as `.on('change')` it breaks the interrupt altogether... it registers the first `change` event but nothing after that.

[gh-152](https://github.com/tessel/t2-firmware/issues/152)